include/deal.II/lac/trilinos_tpetra_vector.templates.h

```diff
@@ -260,15 +277,28 @@ namespace LinearAlgebra
 #  endif
       }
       if (operation == VectorOperation::insert)
-        vector->doExport(source_vector, tpetra_export, Tpetra::REPLACE);
+        vector->doExport(source_vector, *tpetra_export, Tpetra::REPLACE);
       else if (operation == VectorOperation::add)
-        vector->doExport(source_vector, tpetra_export, Tpetra::ADD);
+        vector->doExport(source_vector, *tpetra_export, Tpetra::ADD);
       else
         AssertThrow(false, ExcNotImplemented());
     }
 
 
 
+    template <typename Number>
+    void
+    Vector<Number>::import_elements(
+      const ReadWriteVector<Number> &V,
+      VectorOperation::values        operation,
+      const std::shared_ptr<const Utilities::MPI::CommunicationPatternBase>
+        &communication_pattern)
+    {
+      import_elements(V, operation, Teuchos::rcp(communication_pattern.get()));
```

Comment:
Shouldn't this duplicate the communication pattern? So that the pointer remains valid if/when the `std::shared_ptr` goes out of scope?

Reply:
Here I am not 100% certain,
I assumed, that as long there is a reference to that `std::shared_ptr` the object is not destoryed.

My approach here would be, to create a local object and convert that into a Teuchos::RCP, would that work?

```
    template <typename Number>
    void
    Vector<Number>::import_elements(
      const ReadWriteVector<Number> &V,
      VectorOperation::values        operation,
      const std::shared_ptr<const Utilities::MPI::CommunicationPatternBase>
        &communication_pattern)
    {
      const std::shared_ptr<const Utilities::MPI::CommunicationPatternBase>
        local_comm_pattern =
          std::make_shared<const Utilities::MPI::CommunicationPatternBase>(
            *communication_pattern);

      import_elements(V, operation, Teuchos::rcp(local_comm_pattern.get()));
    }
```